com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/spi/SpiFrameHandler.java

```diff
@@ -311,11 +311,20 @@ private void processSpiCommand(int[] packetData) {
                     logger.debug("<-- RX SPI frame type mismatch SPI_CMD_EZSP != {}",
                             String.format("%02X", lastFrameSent[0]));
                 }
+                
+                // Only trace-log in case of NoCallBacksResponse (which is due to polling)
+                String logMessage = String.format("<-- RX SPI frame: %s", frameToString(packetData));
+                if (EzspFrame.createHandler(Arrays.copyOfRange(packetData, 2, packetData.length)) instanceof EzspNoCallbacksResponse) {
```

Comment:
Instead of this call to ```createHandler``` I wonder if it might not be "better" to peek into the packet and check the appropriate byte for the CallBack Response ID? I know that's not nice as it's breaking the layers, but...
Alternatively, I wonder if we shouldn't copy the data into a temporary buffer to avoid multiple calls to ```copyOfRange```?

Reply:
I had also thought about that, and I cannot really say if either of the two solutions is "better". I will replace by code that "peek into the packet", just to have it explicit (we can still go back if you have some argument that we should not do the "peeking").

Peeking requires to hard-code the frame ID for the NoCallbacksResponse  (`0x07`) here, as the autocoder-generated frame ID constants in the `EzspFrame` class are `protected`.